thunder/dynamo/report.py

```diff
@@ -1042,3 +980,153 @@ def foo(x):
+        graph_name = report.graph_name
+        measure1 = report.run_benchmark(compile_fn1, timer_fn)
+        measure2 = report.run_benchmark(compile_fn2, timer_fn)
+
+        record = False
+        log_strs = ""
+        for m1, m2, name in zip(measure1, measure2, ("forward", "backward")):
+            if m1 is None:
+                assert m2 is None
+                continue
+            ret = check_threshold_log(
+                m1.median, m2.median, compile_fn1.name, compile_fn2.name, f"{graph_name} {name}", timer_name
+            )
+            if not ret[0]:
+                record = True
+                log_strs += f"{ret[1]}\n"
+        if record:
+            extra_comment = f"Benchmark results:\n{log_strs}\n"
+            filename1 = f"{graph_name}_{compile_fn1.name}_{timer_name}.py"
+            filename2 = f"{graph_name}_{compile_fn2.name}_{timer_name}.py"
+            report.write_benchmark(
+                folder_path, compile_fn1, timer_fn, file_name=filename1, extra_comment_str=extra_comment
+            )
+            report.write_benchmark(
+                folder_path, compile_fn2, timer_fn, file_name=filename2, extra_comment_str=extra_comment
+            )
+            print(f"The scripts are saved: {filename1}, {filename2}")
+        print("\n")
+
+    def check_timing_bsym(report, compile_fn1, compile_fn2, timer_fn, timer_name: str):
+        graph_name = report.name
+        measure1 = report.run_benchmark(compile_fn1, timer_fn)
+        measure2 = report.run_benchmark(compile_fn2, timer_fn)
+
+        ret = check_threshold_log(
+            measure1.median, measure2.median, compile_fn1.name, compile_fn2.name, graph_name, timer_name
+        )
+        if not ret[0]:
+            extra_comment = f"Benchmark results:\n{ret[1]}\n"
+            filename1 = f"{graph_name}_{compile_fn1.name}_{timer_name}.py"
+            filename2 = f"{graph_name}_{compile_fn2.name}_{timer_name}.py"
+            report.write_nvfuser_benchmark(folder_path, timer_fn, file_name=filename1, extra_comment_str=extra_comment)
+            report.write_inductor_benchmark(folder_path, timer_fn, file_name=filename2, extra_comment_str=extra_comment)
+            print(f"The scripts are saved: {filename1}, {filename2}")
+        print("\n")
+
+    results = fx_report(fn, *args, compile_options=thunder_compile_kwargs, **kwargs)
+    if thunder_compile_kwargs is None:
+        thunder_compile_kwargs = {}
+    thunderjit_specification = ThunderCompileSpecification(**thunder_compile_kwargs)
+    torchcompile = TorchCompileSpecification()
+    for idx, fx_graph_report in enumerate(results.fx_graph_reports):
+        thunder_fx_graph_report = analyze_thunder_splits(fx_graph_report, **thunder_compile_kwargs)
+        print(f"\n{thunder_fx_graph_report.graph_name}: {thunder_fx_graph_report.split_reason}\n")
+        for split_report in thunder_fx_graph_report.subgraph_reports:
```

Comment:
Very cool; definitely some exciting TODOs here for analyzing the segmentation information and summarizing why segmentation occurred.

A separate TODO is probably making information on the operators used programmatically available. Like being able to query if `torch.add` is in an fx graph, so we can filter for graphs that contain particular calls. And information about operations not executed by thunder, and operations executed by thunder but using "fallback" logic.